addons/account/models/account_payment.py

```diff
@@ -1047,9 +1047,13 @@ def action_post(self):
                     method_name=self.payment_method_line_id.name,
                     partner=payment.partner_id.display_name,
                 ))
-        # Avoid going back one state when clicking on the confirm action in the payment list view and having paid expenses selected
-        # We need to set values to each payment to avoid recomputation later
-        self.filtered(lambda pay: pay.state in {False, 'draft', 'in_process'}).state = 'in_process'
```

Comment:
The point to do that here (outside of the loop) was to be able to batch the write on the state, so that other stuff (like the creation of the journal entries) can be batched too.

Batch is faster because it requires less round trips with the database.